node.go

```diff
@@ -164,6 +172,22 @@ func (node *Node) Visit(opts *Options) (dirs, files int) {
 	return
 }
 
+func (node *Node) match(pattern string, opt *Options) bool {
+	var prefix string
+	if opt.IgnoreCase {
+		prefix = "(?i)"
+	}
+
+	search := node.Name()
+	re, err := regexp.Compile(prefix + pattern)
+	if strings.Contains(re.String(), "*") {
+		search = node.path
+	}
```

Comment:
Can you explain the logic behind this?

Reply:
Firstly it follows the behavior of `tree` which I noted in the description, also if we don't include the path as part of the search it won't match the folder or files on which ever side the wild card character is placed.

```bash
tree -P c --matchdirs /tmp/tree/
/tmp/tree/
└── c
    ├── d
    ├── e
    └── g

tree -P 'c*' --matchdirs /tmp/tree/
/tmp/tree/
└── c
    ├── d
    ├── e
    └── g
        ├── h
        └── i
```

Reply:
Got it. Thanks